dlt/destinations/impl/snowflake/configuration.py

```diff
@@ -80,17 +122,29 @@ def to_connector_params(self) -> Dict[str, Any]:
 
         conn_params: Dict[str, Any] = dict(
             query,
-            user=self.username,
-            password=self.password,
             account=self.host,
             database=self.database,
         )
 
+        if self._use_snowflake_session_token:
+            assert conn_params["authenticator"] == "oauth"
+            conn_params["host"] = self._snowflake_host
+            conn_params["token"] = read_snowflake_session_token()
+        else:
+            conn_params["user"] = self.username
+            conn_params["password"] = self.password
+
         if self.application != "" and "application" not in conn_params:
             conn_params["application"] = self.application
 
         return conn_params
 
+    def _set_use_snowflake_session_token(self) -> None:
+        self.host = os.environ["SNOWFLAKE_ACCOUNT"]
+        self._snowflake_host = os.environ["SNOWFLAKE_HOST"]
+        self.authenticator = "oauth"
```

Comment:
IMO we should ask user to set authenticator to `oauth` explicitly. and only then activate automated token discovery. We do that anyway: ie. if username and host are set we never discover this token. IMO this is confusing. We should tell user that if they set `oauth` and skip `token` OR `host` we'll take it from standard env variables